packages/function-entry/src/hints-popover.jsx

```diff
@@ -0,0 +1,142 @@
+  },
+  'fraction-symbol': {
+    display: 'none',
+  },
+  'radical-symbol': {
+    'text-decoration': 'overline',
+  },
+  power: {
+    'font-size': '12px',
+    'vertical-align': '+50%',
+    'margin-left': '2px',
+  },
+  'hints-popover': {
+    marginTop: theme.spacing.unit * 2,
+    pointerEvents: 'none',
+  },
+  'hints-checkbox': {
+    display: 'inline-block',
+  },
+  'hint-popover-content': {
+    'padding-right': theme.spacing.unit * 2,
+  },
+});
+
+const Fraction = ({ classes, top, bottom }) => (
+  <span className={classes.fraction}>
+      <span><i>{top}</i></span>
+      <span className={`${classes['fraction-span']} ${classes['fraction-symbol']}`}>/</span>
+      <span className={`${classes['fraction-span']} ${classes['fraction-bottom']}`}><i>{bottom}</i></span>
+    </span>
+)
+
+Fraction.propTypes = {
+  classes: PropTypes.object,
+  top: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  bottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const Power = ({ classes, base, exponent }) => (
+  <span>
+      <i>{base}</i>
+      <span className={classes.power}>{exponent}</span>
+    </span>
+);
+
+Power.propTypes = {
+  classes: PropTypes.object,
+  base: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  exponent: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+const Radical = ({ classes, sqrt }) => (
+  <span>
+      &radic;
+    <span className={classes['radical-symbol']}>
+        <i>{sqrt}</i>&nbsp;
+      </span>
+    </span>
+);
+
+Radical.propTypes = {
+  classes: PropTypes.object,
+  sqrt: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+};
+
+class HintsPopover extends React.Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    hintsOpen: PropTypes.bool.isRequired,
+    className: PropTypes.string,
+    hintsAnchorEl: PropTypes.any,
+    anchorReference: PropTypes.any,
+  };
+
+  static defaultProps = {
+    className: '',
+  };
+
+  render() {
+    const { className, classes, hintsOpen, hintsAnchorEl, anchorReference } = this.props;
+
+    return (
+      <Popover
+        className={`${classes['hints-popover']} ${className}`}
+        open={hintsOpen}
+        anchorEl={hintsAnchorEl}
+        anchorReference={anchorReference}
+        onClose={this.handleHintsPopoverClose}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'center',
+        }}
+        disableAutoFocus
+        disableRestoreFocus
+        disableEnforceFocus
+      >
+        <div className={classes['hint-popover']}>
+          <ul>
+            <Typography type="body1" className={classes['hint-popover-content']}>
+              <li>For 2 &#8729; 2 enter 2*2</li>
+              <li>For 3y enter 3y or 3*y</li>
+              <li>For <Fraction top={1} bottom="x" classes={classes} /> enter 1/x</li>
```

Comment:
instead of having to pass `classes` to `Fraction/Power/etc`, why not define them above with classes injected: 

```jsx
const Fraction = withStyles(styles)((props) => ( /*...*/));

//Then: 
<Fraction top={1} bottom="x"/>
```